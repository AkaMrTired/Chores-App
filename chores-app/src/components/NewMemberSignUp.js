/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from "react";

const NewMemberSignUp = () => {
  const initialState = {
    fields: {
      yourName: "",
      password: "",
      confirmPassword: "",
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [passwordError, setPasswordError] = useState();

  const completeAccount = (event) => {
    if (fields.password === fields.confirmPassword) {
      // some axios code to go here to send the field data to the database
      // currently the fields will reset but we can change this so we  change the page to be the parent's home page once this is
      setFields(initialState.fields);
    } else {
      event.preventDefault();
      setPasswordError("Those passwords did not match, please try again");
    }
  };
  const handleFieldChange = (event) => {
    event.preventDefault();
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="sign-up-container">
      <h1>Sign Up</h1>
      <p>
        Please finish creating your account by setting your preferred name and
        password
      </p>
      <div>
        <form onSubmit={completeAccount}>
          <label htmlFor="yourName">Your Name </label>
          <input
            name="yourName"
            required
            type="text"
            placeholder="e.g Mommy"
            value={fields.yourName}
            onChange={handleFieldChange}
          />

          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="password"
            data-testid="password"
            required
            placeholder="*******"
            value={fields.password}
            onChange={handleFieldChange}
          />

          <label htmlFor="confirmPassword">Confirm Password </label>
          <input
            type="password"
            name="confirmPassword"
            required
            placeholder="*******"
            value={fields.confirmPassword}
            onChange={handleFieldChange}
          />
          {!!passwordError && <p>{passwordError}</p>}

          <button type="submit">Complete Account</button>
        </form>
      </div>
    </div>
  );
};

export default NewMemberSignUp;
