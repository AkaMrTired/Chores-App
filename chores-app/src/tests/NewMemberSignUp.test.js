import { render, screen } from "@testing-library/react";
import React from "react";
import NewMemberSignUp from "../components/NewMemberSignUp";

describe("the sign up form", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<NewMemberSignUp />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<NewMemberSignUp />);
    expect(screen.getByText(/Sign Up/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /please finish creating your account by setting your preferred name and password/i
      )
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /Complete Account/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/your name/i)).toBeInTheDocument();
    expect(screen.getByTestId("password")).toBeInTheDocument();
    expect(screen.getByText(/confirm password/i)).toBeInTheDocument();
  });
});
