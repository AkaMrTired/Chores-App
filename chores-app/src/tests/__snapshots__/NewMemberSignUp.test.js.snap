// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the sign up form matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sign-up-container"
  >
    <h1>
      Sign Up
    </h1>
    <p>
      Please finish creating your account by setting your preferred name and password
    </p>
    <div>
      <form>
        <label
          for="yourName"
        >
          Your Name 
        </label>
        <input
          name="yourName"
          placeholder="e.g Mommy"
          required=""
          type="text"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          data-testid="password"
          name="password"
          placeholder="*******"
          required=""
          type="password"
          value=""
        />
        <label
          for="confirmPassword"
        >
          Confirm Password 
        </label>
        <input
          name="confirmPassword"
          placeholder="*******"
          required=""
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Complete Account
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
